controllers/access_token_mapper.go

```diff
@@ -49,7 +50,10 @@ func (at AccessTokenMapper) toSecretType(secretType corev1.SecretType) map[strin
 	case corev1.SecretTypeDockercfg:
 		ret[corev1.DockerConfigKey] = at.Token
 	case corev1.SecretTypeDockerConfigJson:
-		ret[corev1.DockerConfigJsonKey] = at.Token
+		ret[corev1.DockerConfigJsonKey] = fmt.Sprintf(`{"auths":{"%s":{"username":"%s","password":"%s"}}}`,
+			strings.TrimPrefix(at.ServiceProviderUrl, "https://"),
```

Comment:
I'd rather parse the URL and just use the host part of it... We can't be 100% sure the scheme will always be https.